network/network-address/src/lib.rs

```diff
@@ -271,12 +281,39 @@ impl NetworkAddress {
         self.0.push(proto)
     }
 
+    pub fn extend_from_slice(&mut self, protos: &[Protocol]) {
+        self.0.extend_from_slice(protos)
+    }
+
+    /// Given base advertised `NetworkAddress`, append production protocols and
+    /// return the modified `NetworkAddress`.
+    ///
+    /// ### Example
+    ///
+    /// `/dns/example.com/tcp/6180` =>
+    /// `/dns/example.com/tcp/6180/ln-noise-ik/<network_pubkey>/ln-handshake/<handshake_version>`
+    // TODO(philiphayes): handshake version type
+    pub fn into_prod(mut self, network_pubkey: x25519::PublicKey, handshake_version: u8) -> Self {
```

Comment:
hmm... naming is kinda rough, but I can't think of something better ... I wonder how much time you spent here :)